dss/operations/secrets.py

```diff
@@ -291,3 +293,163 @@ def del_secret(argv: typing.List[str], args: argparse.Namespace):
             if not args.quiet:
                 print(f"Secret variable {secret_name} found in secrets manager, deleting it")
             sm_client.delete_secret(SecretId=secret_name)
+
+
+class SecretsChecker(object):
+    """
+    A class to aid in checking deployed secrets in the secrets manager.
+    Will only check canonical HCA DSS stages ('dev', 'integration', 'staging').
+    Checking occurs as follows:
+
+    #1
+    For the json returned from the secret in GOOGLE_APPLICATION_SECRETS_SECRETS_NAME:
+        `auth_uri` should be in ['https://auth.data.humancellatlas.org/authorize',
+                                 'https://auth.dev.data.humancellatlas.org/authorize']
+        `token_uri` should be in ['https://auth.data.humancellatlas.org/oauth/token',
+                                  'https://auth.dev.data.humancellatlas.org/oauth/token']
+
+    #2
+    For the json returned from the secret in GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME:
+        `project_id` should be `human-cell-atlas-travis-test`
+        `type` should be `service_account`
+        `client_email` should be the user account returned from the terraform output "service_account".
+                       For example: dev should be `travis-test@human-cell-atlas-travis-test.iam.gserviceaccount.com`.
+    """
+    def __init__(self, stage):
+        """Initialize variables useful to the secrets checker"""
+        self.stage = stage
+        self.stages = {'dev': 'environment',
+                       'integration': 'environment.integration',
+                       'staging': 'environment.staging'}
```

Comment:
what about `prod` ?

Reply:
This class was migrated from the prior check deployment secrets script, which did not have a prod stage originally either. (I'm not sure if we're running this script on prod.)

Reply:
To clarify: `make deploy` will invoke this script, so it will always run on a `make deploy`, but because no prod stage is defined in `self.stages`, the secrets checker will exit without doing anything when run on the prod stage. (Not sure why that decision was made, it makes sense to me to add the prod stage here.)

Reply:
This was written by me before I had access to `prod`.  I could probably add it in after this is merged in a follow-up PR.